### PR TITLE
8273541: Cleaner Thread creates with normal priority instead of MAX_PRIORITY - 2

### PR DESCRIPTION
**Pull request by kabutz**
…of MAX_PRIORITY-2 during refactoring

Appears in Java 17 for the first time.

During refactoring, the priority was changed from Thread.MAX_PRIORITY -  2 to instead state Thread.MIN_PRIORITY - 2, which results in a negative priority, and is thus set to Thread.NORM_PRIORITY. Thus the Cleaner by default now has threads with priority 5, instead of 8.

The change was done in git revision # 992b50087d2ec8878dfcbbd1820a00b6b6bdf644 and label 8261036 by Claes Redestad

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8273541](https://bugs.openjdk.java.net/browse/JDK-8273541): Cleaner Thread creates with normal priority instead of MAX_PRIORITY - 2

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**)
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5439/head:pull/5439` \
`$ git checkout pull/5439`

Update a local copy of the PR: \
`$ git checkout pull/5439` \
`$ git pull https://git.openjdk.java.net/jdk pull/5439/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5439`

View PR using the GUI difftool: \
`$ git pr show -t 5439`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5439.diff">https://git.openjdk.java.net/jdk/pull/5439.diff</a>

</details>

### CODE REVIEW

**Comment by AlanBateman** [comment]
I've created JDK-8273541 to track this.

File: src/java.base/share/classes/jdk/internal/ref/CleanerImpl.java [hunk 1]
```diff
@@ -225,7 +225,7 @@ public Thread newThread(Runnable r) {
         CleanerCleanable(Cleaner cleaner) {
             super(cleaner, cleaner);
         }
-
+java
```
Comment (shipilev) [inline src/java.base/share/classes/jdk/internal/ref/CleanerImpl.java:228]: Stray change.

**Review by shipilev — CHANGES_REQUESTED** [review]
This looks good. @cl4es might want to confirm that change was not intentional.

File: src/java.base/share/classes/jdk/internal/ref/CleanerImpl.java (hunk 1 above)
Comment (LanceAndersen) [reply src/java.base/share/classes/jdk/internal/ref/CleanerImpl.java:228]: Not sure what the "java" is for here seems like a typo

Comment (kabutz) [reply src/java.base/share/classes/jdk/internal/ref/CleanerImpl.java:228]: Yes, sorry.

**Comment by kabutz** [comment]
Why would jcheck say "This PR contains no changes" @shipilev ?

**Comment by AlanBateman** [comment]
It looks like d621cdd957b78e53ea63b6b9bb2fce08560a8620 has undone the changes.

**Comment by kabutz** [comment]
> It looks like [d621cdd](https://github.com/openjdk/jdk/commit/d621cdd957b78e53ea63b6b9bb2fce08560a8620) has undone the changes.

Thanks Alan!

**Review by shipilev — APPROVED** [review]
This looks fine to me.

**Review by AlanBateman — APPROVED** [review]

**Review by LanceAndersen — APPROVED** [review]
This should be ready to integrate so please enter /integrate in a new comment and then it can be sponsored.

Thank you

**Comment by cl4es** [comment]
> This looks good. @cl4es might want to confirm that change was not intentional.

Completely unintentional, and perplexing since it's not a simple copy-paste error..

**Comment by kabutz** [comment]
> > This looks good. @cl4es might want to confirm that change was not intentional.
> 
> Completely unintentional, and perplexing since it's not a simple copy-paste error..

The thread priorities in Java are the wrong way round, (high is 10, low is 1) so I was not surprised by this mistake.

**Pull request closed without merging**